sport/app/football/feed/Competitions.scala

```diff
@@ -402,4 +405,15 @@ class CompetitionsService(val footballClient: FootballClient, competitionDefinit
     })
     result.map(_.flatten).flatMap(Future.sequence(_))
   }
+
+  def maybeRefreshLiveMatches(
+      clock: Clock,
+  )(implicit executionContext: ExecutionContext): Future[immutable.Iterable[Competition]] = {
+    // matches is the list of all matches from all competitions
+    if (isAMatchInProgress(matches, clock)) {
```

Comment:
We already have the list of matches for all the competitions. it is calculated as `def matches: Seq[FootballMatch] = competitions.flatMap(_.matches)`

Reply:
Careful here, I think this returns **all** the matches for each competition. This means all the fixtures in weeks to come, all the results from previous weeks, and all the currently running matches as well. I've mentioned above in more detail how this affects the logic that's in this change.

Reply:
As I mentioned above, I realised from your other explanation that the date logic I had written was not doing what I expected. I only want to check if the current time is within 5 minutes before the game in case the isLive still has not yet updated. 

But what you explained here, also made me think if we really want to check all the matches or if we may be better off, filtering the list by MatchDay and only check the MatchDay matches. 

Something like this:
```
matches.filter { game =>
      game match {
        case _: MatchDay => true
        case _           => false
      }
    }
```

I saw you mentioned in one of the above comments, that we may wanna use `withTodaysMatches`. We are going to write some test for `withTodaysMatches` since by just looking we were not sure why the `footballMatch.isOn` is doing what it's doing. 

We will get back to you on that, after we write the tests for `withTodaysMatches` and `isAMatchInProgress` :)